examples/storage-disk-bench/src/main.rs

```diff
@@ -0,0 +1,205 @@
+            start_time,
+            size,
+        }
+    }
+}
+
+fn generate_plan(disks_count: u64, activities_count: u64, max_size: u64, max_start_time: u64) -> Vec<DiskRequest> {
+    let mut rnd = CustomRandom::new(SEED);
+
+    let mut plan = vec![];
+
+    // For technical reasons (in SimGrid) first activity should start at time = 0
+    let first_disk_idx = rnd.next() % disks_count;
+    let first_size = rnd.next() % (max_size + 1);
+    plan.push(DiskRequest::new(first_disk_idx as usize, 0, first_size));
+
+    for _ in 0..activities_count - 1 {
+        let disk_idx = rnd.next() % disks_count;
+        let start_time = rnd.next() % (max_start_time + 1);
+        let size = rnd.next() % (max_size + 1);
+        plan.push(DiskRequest::new(disk_idx as usize, start_time, size));
+    }
+
+    plan
+}
+
+impl Runner {
+    fn new(disks: Vec<Rc<RefCell<SharedDisk>>>, ctx: SimulationContext) -> Self {
+        Self {
+            disks,
+            ctx,
+            activities_count: 0,
+            max_size: 0,
+        }
+    }
+
+    fn run(&mut self, activities_count: u64, max_size: u64, max_start_time: u64) {
+        log_info!(self.ctx, "Starting disk benchmark");
+
+        let plan = generate_plan(self.disks.len() as u64, activities_count, max_size, max_start_time);
+
+        self.activities_count = activities_count;
+        self.max_size = max_size;
+
+        for idx in 0..self.activities_count {
+            let request = plan.get(idx as usize).unwrap();
+            self.ctx.emit_self(
+                TimerFired {
+                    disk_idx: request.disk_idx,
+                    size: request.size,
+                },
+                request.start_time as f64,
+            );
+        }
+    }
+
+    fn on_timer_fired(&mut self, disk_idx: usize, size: u64) {
+        self.disks.get(disk_idx).unwrap().borrow_mut().read(size, self.ctx.id());
+    }
+}
+
+impl EventHandler for Runner {
+    fn on(&mut self, event: Event) {
+        cast!(match event.data {
+            DataReadCompleted { request_id: _, size } => {
+                self.activities_count -= 1;
+                log_debug!(
+                    self.ctx,
+                    "Completed reading from {}, size = {}",
+                    self.ctx.lookup_name(event.src),
+                    size,
+                );
+            }
+            DataReadFailed { request_id: _, error } => {
+                log_error!(self.ctx, "Unexpected error: {}", error);
+            }
+            TimerFired { disk_idx, size } => {
+                self.on_timer_fired(disk_idx, size);
+            }
+        })
+    }
+}
+
+fn main() {
+    let args = Args::parse();
+
+    Builder::from_default_env()
+        .format(|buf, record| writeln!(buf, "{}", record.args()))
+        .init();
+
+    let mut sim = Simulation::new(SEED);
+
+    let mut disks = vec![];
+
+    for i in 0..args.disks {
+        let disk_name = DISK_NAME.to_owned() + "-" + &i.to_string();
```

Comment:
```suggestion
        let disk_name = format!("disk-{}", i);
```
и константа `DISK_NAME` не нужна

Reply:
Fixed